### PR TITLE
Fixes GcpRequest comparison for boolean parameters

### LINKED ISSUE #402: gcp_compute_firewall module does not disable firewall rule if the rule exists
##### SUMMARY
When using gcp_compute_firewall module for existing firewall rule, if disabled is changed from "false" to "true", the rule is not disabled (status of the task is "OK" and not "changed", as if no action was necessary).

If I also change something else (priority, ports), the rule is updated and disabled. But if the only thing that is changing is the "disabled" key, the change is ignored.

When the same task and module is used to create a new rule with disabled: true, it works fine.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
gcp_compute_firewall

##### ANSIBLE VERSION
```
ansible 2.9.9
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/pmatuska/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3/dist-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.8.6 (default, Jan 27 2021, 15:42:20) [GCC 10.2.0]

```

##### CONFIGURATION

```

```

##### OS / ENVIRONMENT

Tested locally on Pop!_OS 20.10 
Tested via AWX 17.1.0 running on GKE

##### STEPS TO REPRODUCE
Used following tasks for an existing GCP rule that was enabled:

```
- name: get network info
  gcp_compute_network_info:
    filters:
    - name = <network_name>
    project: <project>
    auth_kind: serviceaccount
    service_account_file: <sa_key_file>
  register: networkInfo

- name: create or update firewall rule
  gcp_compute_firewall:
    name: pm-test-01
    direction: INGRESS
    network: "{{ networkInfo.resources[0] }}"
    project: <project>
    disabled: true
    priority: 900
    denied:
    - ip_protocol: tcp
      ports: ['80', '443']
    - ip_protocol: udp
      ports: 53
    target_tags:
    - proxy
    state: present
    auth_kind: serviceaccount
    service_account_file: <sa_key_file>
```

##### EXPECTED RESULTS
By setting disabled to True, I expected the rule to be disabled.

##### ACTUAL RESULTS

```
TASK [create or update firewall rule] ********************************************************************************************************************************************************
ok: [localhost] => {
    "changed": false,
    "creationTimestamp": "2021-04-14T02:42:41.746-07:00",
    "denied": [
        {
            "IPProtocol": "tcp",
            "ports": [
                "80",
                "443"
            ]
        },
        {
            "IPProtocol": "udp",
            "ports": [
                "53"
            ]
        }
    ],
    "description": "",
    "direction": "INGRESS",
    "disabled": false,
    "id": "<id>",
    "invocation": {
        "module_args": {
            "allowed": null,
            "auth_kind": "serviceaccount",
            "denied": [
                {
                    "ip_protocol": "tcp",
                    "ports": [
                        "80",
                        "443"
                    ]
                },
                {
                    "ip_protocol": "udp",
                    "ports": [
                        "53"
                    ]
                }
            ],
            "description": null,
            "destination_ranges": null,
            "direction": "INGRESS",
            "disabled": true,
            "env_type": null,
            "name": "pm-test-01",
            "network": {
                "autoCreateSubnetworks": false,
                "creationTimestamp": "2021-02-18T05:42:19.082-08:00",
                "id": "<id>",
                "kind": "compute#network",
                "name": "<network>",
                "routingConfig": {
                    "routingMode": "REGIONAL"
                },
                "selfLink": "https://www.googleapis.com/compute/v1/projects/<project>/global/networks/<network>",
                "subnetworks": [
                    "https://www.googleapis.com/compute/v1/projects/<project>/regions/us-central1/subnetworks/<subnet>"
                ]
            },
            "priority": 900,
            "project": "<project>",
            "scopes": [
                "https://www.googleapis.com/auth/compute"
            ],
            "service_account_contents": null,
            "service_account_email": null,
            "service_account_file": "<sa_key_file>",
            "source_ranges": null,
            "source_service_accounts": null,
            "source_tags": null,
            "state": "present",
            "target_service_accounts": null,
            "target_tags": [
                "proxy"
            ]
        }
    },
    "kind": "compute#firewall",
    "logConfig": {
        "enable": false
    },
    "name": "pm-test-01",
    "network": "https://www.googleapis.com/compute/v1/projects/<project>/global/networks/<network>",
    "priority": 900,
    "selfLink": "https://www.googleapis.com/compute/v1/projects/<project>/global/firewalls/pm-test-01",
    "sourceRanges": [
        "0.0.0.0/0"
    ],
    "targetTags": [
        "proxy"
    ]
}
META: ran handlers
META: ran handlers
```

### PR DESCRIPTION
**Pull request by nkakouros**
Fixes #402 .

### CODE REVIEW

File: tests/unit/plugins/test_gcp_utils.py [hunk 1]
```diff
@@ -143,3 +143,34 @@ def test_arrays_dicts_with_difference(self):
...
+        }
+
+        value2 = {
+            "foo": True,
+            "bar": False,
+            "baz": False,
+            "qux": True,
+        }
+
+        value2 = {
```
Comment (toumorokoshi) [inline tests/unit/plugins/test_gcp_utils.py:162]: it looks like this is duplicate?

File: tests/unit/plugins/test_gcp_utils.py [hunk 2]
```diff
@@ -143,3 +143,34 @@ def test_arrays_dicts_with_difference(self):
         request2 = GcpRequest(value2)
         self.assertNotEquals(request1, request2)
         self.assertEqual(request1.difference(request2), difference)
+
+    def test_dicts_boolean_with_difference(self):
```
Comment (toumorokoshi) [inline tests/unit/plugins/test_gcp_utils.py:147]: is it also worth testing with strings too? since "to_text" is used.

**Review by toumorokoshi — CHANGES_REQUESTED** [review]
Thanks for the PR! a couple minor changes and I can merge it!

File: tests/unit/plugins/test_gcp_utils.py (hunk 2 above)
Comment (nkakouros) [reply tests/unit/plugins/test_gcp_utils.py:147]: Could you clarify the use case, what will we be testing? Dictionaries of strings are tested in previous tests.

File: tests/unit/plugins/test_gcp_utils.py (hunk 1 above)
Comment (nkakouros) [reply tests/unit/plugins/test_gcp_utils.py]: Removed.

File: tests/unit/plugins/test_gcp_utils.py (hunk 2 above)
Comment (toumorokoshi) [reply tests/unit/plugins/test_gcp_utils.py:147]: ok, great! that's what I was referring to.

**Review by toumorokoshi — APPROVED** [review]

**Comment by toumorokoshi** [comment]
Great, thanks for the change!

**Pull request merged**